controllers/drill/controller.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2019 The xridge kubestone contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package drill
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	perfv1alpha1 "github.com/xridge/kubestone/api/v1alpha1"
+	"github.com/xridge/kubestone/pkg/k8s"
+)
+
+// Reconciler reconciles a Drill object
+type Reconciler struct {
+	K8S k8s.Access
+	Log logr.Logger
+}
+
+// +kubebuilder:rbac:groups=perf.kubestone.xridge.io,resources=drills,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=perf.kubestone.xridge.io,resources=drills/status,verbs=get;update;patch
```

Comment:
I'm pretty sure some roles are missing here. Configmaps, jobs, etc.

Reply:
the e2e test has succeeded, so we should be covered.